cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorTest.java

```diff
@@ -962,47 +964,396 @@ public void testShouldAutoRestartConnector(VertxTestContext context) {
         KafkaConnectAssemblyOperator op = new KafkaConnectAssemblyOperator(vertx, new PlatformFeaturesAvailability(true, kubernetesVersion),
             supplier, ResourceUtils.dummyClusterOperatorConfig(VERSIONS));
 
-        JsonObject status = new JsonObject();
-        status.put("connector", Map.of("state", "FAILED"));
-
         // Should restart after minute 2 when auto restart count is 1
         var autoRestartStatus =  new AutoRestartStatusBuilder()
             .withCount(1)
             .withLastRestartTimestamp(ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(3).format(DateTimeFormatter.ISO_INSTANT))
             .build();
-        assertThat(op.shouldAutoRestart(autoRestartStatus, status), is(true));
+        assertThat(op.shouldAutoRestart(autoRestartStatus), is(true));
 
         // Should not restart before minute 2 when auto restart count is 1
         autoRestartStatus =  new AutoRestartStatusBuilder()
             .withCount(1)
             .withLastRestartTimestamp(ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(1).format(DateTimeFormatter.ISO_INSTANT))
             .build();
-        assertThat(op.shouldAutoRestart(autoRestartStatus, status), is(false));
+        assertThat(op.shouldAutoRestart(autoRestartStatus), is(false));
 
         // Should restart after minute 12 when auto restart count is 3
         autoRestartStatus =  new AutoRestartStatusBuilder()
             .withCount(3)
             .withLastRestartTimestamp(ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(13).format(DateTimeFormatter.ISO_INSTANT))
             .build();
-        assertThat(op.shouldAutoRestart(autoRestartStatus, status), is(true));
+        assertThat(op.shouldAutoRestart(autoRestartStatus), is(true));
 
         // Should not restart before minute 12 when auto restart count is 3
         autoRestartStatus =  new AutoRestartStatusBuilder()
             .withCount(3)
             .withLastRestartTimestamp(ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(10).format(DateTimeFormatter.ISO_INSTANT))
             .build();
-        assertThat(op.shouldAutoRestart(autoRestartStatus, status), is(false));
+        assertThat(op.shouldAutoRestart(autoRestartStatus), is(false));
 
         // Should not restart after 6 attempts
         autoRestartStatus =  new AutoRestartStatusBuilder()
             .withCount(7)
             .withLastRestartTimestamp(ZonedDateTime.now(ZoneOffset.UTC).minusDays(1).format(DateTimeFormatter.ISO_INSTANT))
             .build();
-        assertThat(op.shouldAutoRestart(autoRestartStatus, status), is(false));
+        assertThat(op.shouldAutoRestart(autoRestartStatus), is(false));
 
         context.completeNow();
     }
 
+    @Test
```

Comment:
You could do this with a parametrized test if you wanted to have different tests for each case. It would potentially make it easier to see which one failed and less code overall, but would mean it instantiates the operator each time.

Reply:
Yeah, it is basically a copy-paste of the previous test. I think it is fine the way it is - its just a few variations.